systemverilog-plugin/UhdmAst.cc

```diff
@@ -168,6 +168,18 @@ static void delete_children(AST::AstNode *node)
     node->children.clear();
 }
 
+static std::vector<AST::AstNode *> get_ranges(AST::AstNode *id)
+{
+    log_assert(!id->children.empty());
+    std::vector<AST::AstNode *> ranges;
+    for (auto c : id->children) {
+        if (c->type == AST::AST_RANGE) {
+            ranges.push_back(c);
```

Comment:
From what I see this function copies pointers, right?
I think we should either `clone` the nodes, or, If we don't need to access them from the original node, erase the pointers from `id->children` after the pointers are copied to a vector to be returned from the function.